JamGame/Assets/Scripts/TileUnion/PlaceCondition.cs

```diff
@@ -8,10 +8,37 @@
 
 namespace TileUnion.PlaceCondition
 {
+    public class Result
+    {
+        public bool Success { get; protected set; }
+        public bool Failure => !Success;
+
+        protected List<TileImpl> FailedTiles_ = null;
+        public List<TileImpl> FailedTiles =>
+            Success ? throw new Exception("Condition pass, no failed tiles.") : FailedTiles_;
+    }
+
+    public class SuccessResult : Result
+    {
+        public SuccessResult()
+        {
+            Success = true;
+        }
+    }
+
+    public class FailResult : Result
+    {
+        public FailResult(List<TileImpl> failedTiles)
+        {
+            FailedTiles_ = failedTiles;
+            Success = false;
+        }
+    }
```

Comment:
This is redundant and still confusing. Better use approach described here https://github.com/mertwole/miniature-happiness/pull/25#discussion_r1284073706